index.bs

```diff
@@ -434,11 +438,18 @@ XRWebGLDepthInformation {#xr-gpu-depth-info-section}
 [Exposed=Window]
 interface XRWebGLDepthInformation : XRDepthInformation {
   [SameObject] readonly attribute WebGLTexture texture;
+
+  readonly attribute XRTextureType textureType;
+  readonly attribute unsigned long? imageIndex;
 };
 </script>
 
 The {{XRWebGLDepthInformation/texture}} attribute contains depth buffer information as an [=opaque texture=]. Each texel corresponds to distance from the [=XRDepthInformation/view=]'s near plane to the users' environment, in unspecified units. The size of each data entry and the type is determined by {{XRSession/depthDataFormat}}. The values can be converted from unspecified units to meters by multiplying them by {{XRDepthInformation/rawValueToMeters}}. The {{XRDepthInformation/normDepthBufferFromNormView}} can be used to transform from [=normalized view coordinates=] into depth buffer's coordinate system. When accessed, the algorithm to <a lt="attempt to access the depth buffer">access the depth buffer</a> of {{XRDepthInformation}} MUST be run.
 
+The {{XRWebGLDepthInformation/textureType}} attribute describes if the texture is of type {{TEXTURE_2D}} or {{TEXTURE_2D_ARRAY}}.
```

Comment:
I'd like there to be some more concrete information about how the texture type is selected. The text in this change makes it sound like the OS chooses for you and you have no say in the matter, but that's not how we handle the texture type selection anywhere else. For example, we allow developers to specify the texture type for XRLayers.

Feels like this would ideally to be linked with the developers use of projection layers or specified with the `XRDepthStateInit` (which is at session creation time, so possibly too restrictive?) I'd like developers to never be surprised by the texture type they received.

I recognize that in some cases that might mean that an internal copy would have to happen, and that would be unfortunate, but it might also make the difference between an Mobile-tested experience "just working" on a headset or the entire app failing to render because of a texture type mismatch.

Reply:
It would be unfortunate if we force implementators to do a slow copy behind the scenes to get a side-by-side texture. Maybe it's better to drop the regular texture and always return a texture-array if there's a chance for developer confusion.

Reply:
I'm concerned that would break any existing mobile AR apps that use it? But I guess that depends on WebGL's behavior around texture binding. I need to refresh my memory as to how that works, but don't you have to bind to the `GL_TEXTURE_2D_ARRAY` to interact with the texture and use the `sampler2DArray` type in the shader to sample from it. I don't think you could treat a texture array with a single layer as a `GL_TEXTURE_2D` and have it work seamlessly.

Reply:
> I'm concerned that would break any existing mobile AR apps that use it?

AFAIK we're the first ones to offer GPU acceleration. Android uses the software path.

> But I guess that depends on WebGL's behavior around texture binding. I need to refresh my memory as to how that works, but don't you have to bind to the `GL_TEXTURE_2D_ARRAY` to interact with the texture and use the `sampler2DArray` type in the shader to sample from it. I don't think you could treat a texture array with a single layer as a `GL_TEXTURE_2D` and have it work seamlessly.

Correct. However, a texture array with 1 element is still fine.

Reply:
> AFAIK we're the first ones to offer GPU acceleration. Android uses the software path.

If that's the case it changes my opinion, and I'd be OK with it always being a texture array. @bialpio, is this accurate?